### ISSUE TITLE
New Garmin update seems to have broken my beloved Caltrack

### ISSUE THREAD

**Issue by SybilFenix3** [CodyJung/connectiq-apps#7]
I have the Fenix3 HR.  I adore this  app and have used it every day since I downloaded (I reviewed it with 5 stars and a happy dance in Jan of this year. . . Username "LittleBad" on GarminConnect)).  But since the latest Garmin software 'upgrade', I am unable to access History. I get simply the "IQ !" icon,, showing something being pulled has an error. I have tried an un-install and re-install of the app to no avail.  I know this is not your issue, Cody, but being the expert - I am REALLY hoping you can help advise on a fix, other than the uninstall/re-install I have tried.

Thank you!

**Comment by CodyJung** [comment]
Oh no! I'm not super surprised, since the CIQ 1.3.x update was pretty substantial. I'll update my fenix and give it a shot. Hopefully just rebuilding it will help.

**Comment by SybilFenix3** [comment]
Thank you so much for you reply!  Let me know how it goes, and what I should mimmick if you get it tamed.

Thanks!!

**Comment by CodyJung** [comment]
Uploaded version 1.3.0 with this fix to the store! You should be able to download it and give it a shot shortly (depending on how long it takes the app to get rolled out).

**Comment by SybilFenix3** [comment]
Thank you !   . . . but Arrrgghh!

So - First let me say THANK YOU for working on this so quickly . . you rock  . . .and I almost feeling guilty reporting back with a not “yippee”,  but for some reason mine is still not working.

AT first the auto-update hung , so , to be sure I can getting a clean copy, I completely un-installed and re-installed (and “cleared History”) before contacting again . . .  (twice, actually,  because on the second round , I tried a “download” straight from the Connect IQ website instead of simply allowing app Manager on Garmin Express to handle)

That’s all the variant tricks I could come up with on this side. 

I know this came from Garmin’s update , and not you app . . . but any additional advice ?

Thank you (again!)

Sybil

**Comment by SybilFenix3** [comment]
Thank you !   . . . but Arrrgghh!

So - First let me say THANK YOU for working on this so quickly . . you rock  . . .and I almost feeling guilty reporting back with a not “yippee”,  but for some reason mine is still not working.

AT first the auto-update hung , so , to be sure I can getting a clean copy, I completely un-installed and re-installed (and “cleared History”) before contacting again . . .  (twice, actually,  because on the second round , I tried a “download” straight from the Connect IQ website instead of simply allowing app Manager on Garmin Express to handle)

That’s all the variant tricks I could come up with on this side. 

I know this came from Garmin’s update , and not you app . . . but any additional advice ?

Thank you (again!)
Sybil

**Comment by CodyJung** [comment]
Ach, that's super strange. I've got working on my watch, so I was hoping that was the issue.  Is it still the same problem (crashing while trying to access history?)

I've attached the PRG from my watch. If you can:
1. Remove the PRG from the APPS directory on your watch
2. Unplug your watch so it completely removes the app
3.. Put this PRG on instead:
https://www.dropbox.com/s/n8jzjwfpjses4tz/caltrack.prg?dl=1

If this doesn't work, I'll get you a different debug build that should hopefully help narrow down the issue.

**Comment by SybilFenix3** [comment]
Hi Cody, 

As I become more and more in your debt while you work with me (and thank you!). I still cannot report success.

Just to make sure I followed your request as you expected . . I listed my steps here
1) While plugged in , used App Manager (Garmen Express, or "GE", later) to remove CalTrack App
2) Confirmed the current program (D8C699A3.prg) disappeared from the APPS\ folder. (and I assume it was the correct one because it was the only one with download date of today)
3) Unplugged and confirmed no Caltrack on watch
4) downloaded Caltrack.prg from dropbox 
5) Plugged back in and pasted Caltrack.prg RIGHT into the (External drive APPS\ folder 
6) Sync'd (by the way, after syncing I checked the 'Caltrack'
![history error](https://cloud.githubusercontent.com/assets/21007865/17649930/1abf857a-620e-11e6-914b-8dc62e314713.jpg)
 status on App manager in GE and it still appeared to be inactive ('download' button was showing, rather than the "X") . .but  when I 
7) Unplugged and checked for app it was indeed there, but  . .
8) When I tried to check History I still got only the IQ! screen (RATS And CATS)

By the way - PLEASE don't feel obligated to keep working this same-day. I will be thrilled to get a resolution, but I certainly don't want to muck up your weekend, or be too greedy with your  time.

Thanks again!
Sybil

**Comment by SybilFenix3** [comment]
Hi Cody, 

As I become more and more in your debt while you work with me (and thank you!). I still cannot report success.

Just to make sure I followed your request as you expected . . I listed my steps here

1) While plugged in , used App Manager (Garmen Express, or "GE", later) to remove CalTrack App

2) Confirmed the current program (D8C699A3.prg) disappeared from the APPS\ folder. (and I assume it was the correct one because it was the only one with download date of today)

3) Unplugged and confirmed no Caltrack on watch

4) downloaded Caltrack.prg from dropbox 

5) Plugged back in and pasted Caltrack.prg RIGHT into the (External drive APPS\ folder 

6) Sync'd (by the way, after syncing I checked the 'Caltrack’ status on App manager in GE and it still appeared to be inactive ('download' button was showing, rather than the "X") . .but  when I 

7) Unplugged and checked for app it was indeed there, but  . .

8) When I tried to check History I still got only the IQ! screen (RATS And CATS)

By the way - PLEASE don't feel obligated to keep working this same-day. I will be thrilled to get a resolution, but I certainly don't want to muck up your weekend, or be too greedy with your  time.

Thanks again!

Sybil

**Comment by SybilFenix3** [comment]
WOAH!!  Sorry that pic showed up to HUGE here!!   I resized and thought it would be more reasonable. (feel free to delete  if you can edit comments)

**Comment by CodyJung** [comment]
Got it! Apparently you need to add at least one menu item now, so I added a case for "no history".

**Comment by SybilFenix3** [comment]
WOOOOOHOOOOOOOOOO!!!

Yes! I see the ‘No History’ ! So I think I am GOOD . .but will still confirm tomorrow that I can see yesterday’s history and all is well.  (and my apologies – so I gather correctly that if I had just waited until tomorrow and there WAS History, form Sat, that is would have been all good, and saved you the last chunk of time ?)

For the record I did wait a day on my initial realization .  . . .  but this time all my checks were, as you know, same day.

Anyway – WHEEEEEEE!!

Thank you, Thank you, Thank you!!!

I wish they’d let me do another review so I could give you another 5-stars!! (or send you a e-drink  . . or something . . actually if I knew your regular address I would send you an e-drink . . your call J)

Thanks again!

I’m going to go eat something . . . just to log it, LOL.

Cheers.

-Sybil
